asserts/account_key.go

```diff
@@ -96,6 +104,21 @@ func (ak *AccountKey) checkConsistency(db RODatabase, acck *AccountKey) error {
 	if err != nil {
 		return err
 	}
+	// XXX: Make this unconditional once account-key assertions are required to have a name.
+	if ak.Name() != "" {
+		assertions, err := db.FindMany(AccountKeyType, map[string]string{
+			"account-id": ak.AccountID(),
+			"name":       ak.Name(),
+		})
+		if err == nil {
+			for _, assertion := range assertions {
+				existingAccKey := assertion.(*AccountKey)
+				if ak.PublicKeyID() != existingAccKey.PublicKeyID() {
+					return fmt.Errorf("new account-key assertion for %q (%q) has ID %q, but clashes with existing ID %q", ak.AccountID(), ak.Name(), ak.PublicKeyID(), existingAccKey.PublicKeyID())
```

Comment:
I don't think the error message really explains *what* clashed. Maybe "account-key assertion for {account-id} with ID {new key's id} has the same name, {name}, as existing ID {existing key id}"?